### ISSUE TITLE
error: Building for iOS, but the embedded framework 'Instabug.framework' was built for iOS + iOS Simulator

### ISSUE THREAD

**Issue by avargaskun** [Instabug/instabug-cordova#113]
I am getting the following error when attempting to build using the latest version of the instabug-cordova plugin:

`error: Building for iOS, but the embedded framework 'Instabug.framework' was built for iOS + iOS Simulator`

My version of `xcode` is `12.3`
My version of `cordova` is `10.0.0`
My version of `ios` cordova platform is `6.1.1`
My version of `instabug-cordova` is `^9.1.6`

I get the same error when building from xcode instead of using the cordova cli.

I found some information online which would indicate the Instabug framework would need to be recompiled with separate binaries for iOS and Simulator. Does that ring a bell? Source: https://developer.apple.com/forums/thread/130684?answerId=413990022#413990022

**Comment by AliAbdelfattah** [comment]
Hi @avargaskun, this is probably happening because the stripping script for simulator binaries wasn't executed for some reason. Check the Build Phases in XCode, there should be a build phase with the name `StripFrameworkScript`, if it's not there, you can either [add it manually](https://github.com/Instabug/instabug-cordova/blob/4d6efa8aa21a134f2d42bfa1eda1b6c77b077580/scripts/zStripScript.js#L61) or you could try reinstalling our plugin (the build phase automatically gets added post-installation), and then check XCode build phases again.

**Comment by MarioC3** [comment]
Hi @avargaskun I also have the same problem and I do have the stripping script in Xcode Build Phases. Also the same Stack. Any other ideas you think it could be affecting this?

**Comment by EvelinJK** [comment]
Same problem here @AliAbdelfattah. 
All our devs have this issue now after we upgraded from  XCode 12.1 to 12.3. 
Any new info as how to fix it?

**Comment by bsameh-at-instabug** [comment]
@MarioC3 @EvelinJK my sincere apologies for not getting back to you earlier and thank you for your patience.

To give you some context, with the release of Xcode 12.3, Apple is now enforcing the use of XCFramework, so we've updated our Cordova SDK to depend on v10.1.1 of our iOS SDK instead, which adds support to that format. We've pushed a custom version of the SDK so you can give it a try, which you can install via the following command:
`cordova plugin add https://github.com/Instabug/instabug-cordova.git\#hotfix/xcframework`

Please let us know if this addresses your issue, we will be waiting for your feedback.

**Comment by MarioC3** [comment]
Hi @bsameh-at-instabug
Thanks for responding. With the new SDK I can build from the Command Line, but whenever I try to archive from XCode, I'm getting this error:

```
/Users/marioacarlos/tallLocal/embark/platforms/ios/Embark/Plugins/instabug-cordova/IBGPlugin.m:2:9: fatal error: 'Instabug/Instabug.h' file not found
#import <Instabug/Instabug.h>
        ^~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

**Comment by bsameh-at-instabug** [comment]
For anyone running into the same issue, specifically those building the app through Xcode itself, please make sure to open the `.xcworkspace` file and not the `.xcodeproj` one. This is because the native SDK is imported using Cocoapods now, which generates a workspace file with the dependencies (pods) included in a separate project within that workspace, hence why it couldn't find the header file. 

@MarioC3 @EvelinJK I apologize again for the delay and I hope your issue has been resolved. Will be closing this as soon as the change is merged to `master`.

**Comment by EvelinJK** [comment]
> @MarioC3 @EvelinJK my sincere apologies for not getting back to you earlier and thank you for your patience.
> 
> To give you some context, with the release of Xcode 12.3, Apple is now enforcing the use of XCFramework, so we've updated our Cordova SDK to depend on v10.1.1 of our iOS SDK instead, which adds support to that format. We've pushed a custom version of the SDK so you can give it a try, which you can install via the following command:
> `cordova plugin add https://github.com/Instabug/instabug-cordova.git\#hotfix/xcframework`
> 
> Please let us know if this addresses your issue, we will be waiting for your feedback.

Thanks for the info @bsameh-at-instabug
The hotfix removes the original error for us. But it lead to issues with other plugins that need pods. And even after we fixed that, we get errors after deploy to a device that break our application due to instabug: 
`.... .app/Frameworks/Instabug.framework/Instabug: mach-o, but not built for platform iOS ...`
`...app/Frameworks/Instabug.framework/Instabug: stat() failed with errno=25`

We're still investigating what the issues could be.
But atm we're unable to use Instabug anymore.

**Comment by bsameh-at-instabug** [comment]
@EvelinJK sorry to hear that, we will look into this right away. Do you mind sharing the iOS version of the device you're using?

**Comment by EvelinJK** [comment]
@bsameh-at-instabug 
Thanks for the quick reply.

iOS version on device is `14.3`
XCode `12.3`
cordova `10.0.0`
cordova-ios: `^5.1.1` (stuck with this atm)

To give more information after we tried various combinations now: 

We checked again and the core issue is that the instabug framework seems not to get embedded correctly with the hotfix. It's missing from the Frameworks list in XCode General tab after building via command line. 

The binary builds fine though and we don't get any errors from that. After deployment to the device via XCode the above error than occurs.

With version `9.1.6` of the plugin and XCode `12.2` the framework gets embedded just fine and we can build successfully via command line to the device. 
But we are unable to upload the binary to Testflight since the instabug framework causes errors like `contains unsupported architectures ...` and `invalid segment alignment..` ` ...try rebuilding with the latest XCode version` 😅  ... which we can't due to the original error reported in this thread. So we're kinda stuck now.

**Comment by bsameh-at-instabug** [comment]
@EvelinJK thank you for the detailed breakdown, this is very helpful! We're looking into this and will get back to you as soon as possible.

**Comment by bsameh-at-instabug** [comment]
@EvelinJK just to confirm, did you remove the existing plugin via `cordova plugin remove instabug-cordova` before installing the new custom build via the command I shared [here](https://github.com/Instabug/instabug-cordova/issues/113#issuecomment-763920454)?

**Comment by EvelinJK** [comment]
@bsameh-at-instabug yes, we removed the old one beforehand and did a clean install.

**Comment by AliAbdelfattah** [comment]
Hi @EvelinJK, unfortunately, we haven't been able to reproduce the issue at our end. Do you mind jumping on a call to debug it? Please head to [Instabug](https://dashboard.instabug.com) and press on the chat button at the right bottom corner, our team will take it from there to set up the call.

**Comment by EvelinJK** [comment]
@AliAbdelfattah Thank's for all your effort :)
We will investigate again if we can get it to work somehow and report here if we find a solution. 
If we're still unable to fix it, we'll get in touch with you as you recommended.

**Comment by EvelinJK** [comment]
@AliAbdelfattah
Good news from our side: We took a look at it again and are now able to release with the custom SDK version provided by you in this thread. We didn't find any specifics, but did the following: 

Update system to `Big Sur 11.1`
Update to latest `XCode 12.4` 
remove and re-install all node modules
remove and re-install Cocoa Pods with latest version `1.10.1`
update other plugins that use pods

We are now able to build and upload via fastlane and XCode again 🎉 .

Not sure what the exact issue was. But we had many problems before in our project when using multiple plugins that need cocoa pods. But 🤷‍♀️ ... Always a mystery.

Do you know when/if the hotfix will be merged into master?

**Comment by AliAbdelfattah** [comment]
[Fix](https://github.com/Instabug/Instabug-Cordova/issues/113#issuecomment-763920454) merged and released.
